### PR TITLE
fix: Delete cronjobs when pruning disabled and actually reconcile scheduler corncobs

### LINKED ISSUE #337: Disabling DB pruning does not delete CronJobs
Should have a PR soon.

### PR DESCRIPTION
**Pull request by Sovietaced**
Fixes #337 

* Reconcile scheduler pruner cron jobs
* Update logic to delete previously enabled cron jobs when pruning is disabled. Otherwise, disabling pruning does nothing and the jobs will continue to run.

## Type of change

Please select the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code Style Update (formatting, renaming)
- [ ] Refactor (code changes that do not fix a bug or add a feature)
- [ ] Documentation Update
- [ ] Other (please describe):

## How Has This Been Tested?
See unit tests. 

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: internal/controller/install/common_helpers.go [hunk 1]
```diff
@@ -596,13 +596,13 @@ func upsertObjectIfNeeded(
 	mutateFn controllerutil.MutateFn,
 	logger logr.Logger,
 ) error {
-	if !isNil(object) {
-		logger.Info(fmt.Sprintf("Upserting %s %s object", componentName, object.GetObjectKind()))
-		if _, err := controllerutil.CreateOrUpdate(ctx, client, object, mutateFn); err != nil {
-			return err
-		}
+	if isNil(object) {
```
Comment (Sovietaced) [inline internal/controller/install/common_helpers.go:599]: Small style cleanup so the majority of the code doesn't have to get nested in an `if` condition

File: internal/controller/install/common_helpers.go [hunk 2]
```diff
@@ -619,6 +619,32 @@ func isNil(i any) bool {
...
+	componentName string,
+	logger logr.Logger,
+) error {
+	if isNil(object) {
+		return nil
+	}
+
+	err := client.Delete(ctx, object)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
```
Comment (dejanzele) [inline internal/controller/install/common_helpers.go:636]: This whole `if/else` block can be replaced with the `return client.IgnoreNotFound(err)`

File: internal/controller/install/common_helpers.go [hunk 3]
```diff
@@ -619,6 +619,32 @@ func isNil(i any) bool {
...
+	client client.Client,
+	object client.Object,
+	componentName string,
+	logger logr.Logger,
+) error {
+	if isNil(object) {
+		return nil
+	}
+
+	err := client.Delete(ctx, object)
```
Comment (dejanzele) [inline internal/controller/install/common_helpers.go:634]: I'd move the `err` in the `if` init block below

File: internal/controller/install/common_helpers.go (hunk 2 above)
Comment (Sovietaced) [reply internal/controller/install/common_helpers.go]: Sounds good 4238007b3559cf8290d01ebb79a96acec350596f

**Review by dejanzele — APPROVED** [review]

**Pull request merged**